provisioning/baremetal_webhook.go

```diff
@@ -0,0 +1,160 @@
+package provisioning
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pkg/errors"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+)
+
+const (
+	validatingWebhookService           = "baremetal-operator-webhook-service"
+	validatingWebhookConfigurationName = "baremetal-operator-validating-webhook-configuration"
+	validatingWebhookServiceHttpsPort  = 443
+	validatingWebhookHttpPortName      = "http"
+)
+
+// EnsureBaremetalOperatorWebhook ensures ValidatingWebhook resources are ready to serve.
+func EnsureBaremetalOperatorWebhook(info *ProvisioningInfo) (bool, error) {
+	if !info.BaremetalWebhookEnabled {
+		// In case service-ca operator is degraded and we detect it in reconciliation loop
+		// We should delete validatingwebhook resources(ValidatingWebhookConfiguration, Service)
+		// to allow BMO continues serving. If we detect in first run, it is safe to run these
+		// deletions.
```

Comment:
I seriously doubt that any other operator does this. I'd suggest just leave it in place.